myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+                ):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
+                else:  # Error occurred
+                    error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+                    error = error.replace("Description ", "")  # For cosmetic reasons
+                    raise MyfitnesspalRequestFailed(
+                        f"Unable to submit food to MyFitnessPal: {error}"
+                    )
+            except Exception:
+                logger.warning(f"Unable to submit food to MyFitnessPal: {error}")
+
+        elif not result.ok:
+            logger.warning(
+                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+            )
+            return None
+
+    def set_new_goal(
+        self,
+        energy: float = None,
+        energy_unit: str = None,
+        carbohydrates: float = None,
+        protein: float = None,
+        fat: float = None,
+        percent_carbohydrates: float = None,
+        percent_protein: float = None,
+        percent_fat: float = None,
+        # TODO Add micro nutritions
+        # saturated_fat: float = None,
+        # polyunsaturated_fat: float = None,
+        # monounsaturated_fat: float = None,
+        # trans_fat: float = None,
+        # fiber: float = None,
+        # sugar: float = None,
+        # cholesterol: float = None,
+        # sodium: float = None,
+        # potassium: float = None,
+        # vitamin_a: float = None,
+        # vitamin_c: float = None,
+        # calcium: float = None,
+        # iron: float = None,
+        assign_exercise_energy="nutrient_goal",
+    ):
+        """Function to update your nutrition goals. Function will return True if successful."""
+        # FROM MFP JS:
+        # var calculated_energy = 4 * parseFloat(this.get('carbGrams')) + 4 * parseFloat(this.get('proteinGrams')) + 9 * parseFloat(this.get('fatsGrams'));
+
+        # Get User Default Unit Preference
+        if energy_unit != "calories" and energy_unit != "kilojoules":
+            energy_unit = self.user_metadata["unit_preferences"]["energy"]
+
+        # Get authenticity token and current values
+        url = parse.urljoin(self.BASE_URL_SECURE, "account/my_goals")
+        today = datetime.datetime.now().strftime("%Y-%m-%d")
+
+        """
+        Will leave this here as reminder why we do not use self._get_json_for_url() at the moment.
+        # No Way implemented to send token by calling self._get_json_for_url
+        old_goals_url = parse.urljoin(self.BASE_API_URL, f"v2/nutrient-goals?date={today}")
+        old_goals = self._get_json_for_url(old_goals_url)
+        """
```

Comment:
```suggestion
        # Will leave this here as reminder why we do not use self._get_json_for_url() at the moment.
        # No Way implemented to send token by calling self._get_json_for_url
        # old_goals_url = parse.urljoin(self.BASE_API_URL, f"v2/nutrient-goals?date={today}")
        # old_goals = self._get_json_for_url(old_goals_url)
```

I'm not quite sure I understand what the comment is explaining, though -- I wonder if you would mind elaborating on what it's trying to remind future-us about?

Reply:
Sure. to get a proper response from MFP we need to send additional headers with our post to the BASE_API_URL.
But it is not implemented in _get_json_for_url() to pass these to the function.

But in the end it does not really matter as it would only save 1 or 2 lines. I will remove it to keep the code clean.